TUnit.Core.SourceGenerator/CodeGenerators/Formatting/TypedConstantFormatter.cs

```diff
@@ -386,10 +387,46 @@ private bool AreValuesEqual(object? enumValue, object? providedValue)
 
     private static string EscapeForTestId(string str)
     {
-        return str.Replace("\\", "\\\\")
-                  .Replace("\r", "\\r")
-                  .Replace("\n", "\\n")
-                  .Replace("\t", "\\t")
-                  .Replace("\"", "\\\"");
+        var needsEscape = false;
+        foreach (var c in str)
+        {
+            if (c == '\\' || c == '\r' || c == '\n' || c == '\t' || c == '"')
+            {
+                needsEscape = true;
+                break;
+            }
+        }
```

Comment:
since .NET 8 it is possible to do this: `var needsEscape = str.AsSpan().ContainsAny(['\\', '\r', '\n', '\t', '"']);`